ee/localserver/dt4a.go

```diff
@@ -78,3 +88,46 @@ func (ls *localServer) requestDt4aInfoHandlerFunc(w http.ResponseWriter, r *http
 	w.Header().Set("Content-Type", "application/json")
 	w.Write(dt4aInfo)
 }
+
+func (ls *localServer) requestDt4aAccelerationHandler() http.Handler {
+	return http.HandlerFunc(ls.requestDt4aAccelerationHandlerFunc)
+}
+
+func (ls *localServer) requestDt4aAccelerationHandlerFunc(w http.ResponseWriter, r *http.Request) {
+	r, span := traces.StartHttpRequestSpan(r, "path", r.URL.Path)
+	defer span.End()
+
+	// Validate origin. We expect to either have the origin set to an allowlisted value, or to be
+	// present but empty, or to be missing. We will not allow a request with a nonempty origin
+	// that is not in the allowlist.
+	requestOrigin := r.Header.Get("Origin")
+	if requestOrigin != "" {
+		if _, ok := allowlistedDt4aOriginsLookup[requestOrigin]; !ok && !strings.HasPrefix(requestOrigin, safariWebExtensionScheme) {
+			escapedOrigin := strings.ReplaceAll(strings.ReplaceAll(requestOrigin, "\n", ""), "\r", "") // remove any newlines
+			ls.slogger.Log(r.Context(), slog.LevelInfo,
+				"received dt4a request with origin not in allowlist",
+				"req_origin", escapedOrigin,
+			)
+			w.WriteHeader(http.StatusForbidden)
+			return
+		}
+	}
```

Comment:
@James-Pickett let me know if you want me to move these origin checks into the middleware now -- I went with this to start because I didn't want to cause merge conflicts if you've got something in flight

Reply:
I don't have any dt4a stuff in flight at the moment, feel free to move it over.